airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py

```diff
@@ -363,3 +364,17 @@ def is_greater_than_or_equal(self, first: Record, second: Record) -> bool:
             return True
         else:
             return False
+
+    def set_runtime_lookback_window(self, lookback_window_in_seconds: int) -> None:
```

Comment:
Neither CustomIncrementalSync nor the Cursor class enforces the lookback parameter, which means that GlobalSubstream has a dependency on DateTimeBasedCursor. 
If we decide to change the value of the lookback window in the model instead of the component, we would need to add logic for parsing and comparing the lookback window directly within GlobalCursor. This approach would require additional logic to handle any new lookback formats that might be introduced, as each new format would necessitate specific parsing and lookback window logic in GlobalCursor, rather than in the respective cursor class.

Reply:
Regarding the top comment about the parent stream, I have [added the test with multiple levels of parents](https://github.com/airbytehq/airbyte/pull/39593/files#diff-4ccba36c75568c24343e4947377954f691a5831ac98d0367faf9da63b966a51cR972-R984), or what do you mean?

Reply:
> which means that GlobalSubstream has a dependency on DateTimeBasedCursor

That makes sense to me. Should we make that explicit in the `GlobalSubstream.__init__`? Right now, the interface accepts any cursor.

> Regarding the top comment about the parent stream, I have [added the test with multiple levels of parents](https://github.com/airbytehq/airbyte/pull/39593/files#diff-4ccba36c75568c24343e4947377954f691a5831ac98d0367faf9da63b966a51cR972-R984), or what do you mean?

You are right, I misread part of the code which led me to think that the cursor passed to the global stream object was the parent cursor but it's actually always the cursor defined in the incremental_sync